src/figuresClasses.ts

```diff
@@ -1,19 +1,81 @@
 export interface Figure {
-
+  getArea(): number | Error
 }
 
-export class Triangle {
+export enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
 
+export enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('one of sides is equal 0 or less');
+    }
+
+    if (a >= b + c || b >= c + a || c >= b + a) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
+    }
+  }
+
+  getArea(): number | Error {
+    const { a } = this;
+    const { b } = this;
+    const { c } = this;
 
+    const s: number = (a + b + c) / 2;
```

Comment:
By default we should rely as much as possible on type inference. Here typescript by default knows that `s` will be a number, there is no point in typing it explicitly.

Reply:
and `getArea()` if you type it on the interface and make classes implement it, there is no need for further typing